src/main/groovy/org/unbrokendome/gradle/plugins/testsets/internal/TestTaskListener.groovy

```diff
@@ -32,10 +32,8 @@ class TestTaskListener {
             if (testTask.metaClass.respondsTo(testTask, "getTestClassesDirs")) {
                 // Gradle 4.0+
                 // See https://docs.gradle.org/4.0/release-notes.html#detecting-test-classes-for-custom-test-tasks
-                map('testClassesDirs') {
-                    def sourceSet = (SourceSet) project.sourceSets[testSet.sourceSetName]
-                    sourceSet.output.classesDirs
-                }
+                def sourceSet = project.sourceSets[testSet.sourceSetName]
+                testTask.testClassesDirs += sourceSet.output.classesDirs
```

Comment:
On it. @tkrullmann do you want a comment in there to explain why no convention mapping for gradle 4+?